src/exaerror.lua

```diff
@@ -0,0 +1,164 @@
+local M = {
+    VERSION = "1.0.0",
+}
+
+-- Lua 5.1 backward compatibility
+_G.unpack = table.unpack or _G.unpack
+
+---
+-- Convert error to a string representation.
+-- <p>
+-- Note that <code>__tostring</code> is the metamethod called by Lua's global <code>tostring</code> function.
+-- This allows using the error message in places where Lua expects a string.
+-- </p>
+--
+-- @return string representation of the error object
+--
+function M:__tostring()
+    local pieces = {}
+    if self.message then
+        table.insert(pieces, self:get_message())
+    end
+    if self.mitigations then
+        for _, mitigation in ipairs(self.mitigations) do
+            table.insert(pieces, mitigation)
```

Comment:
In Java for multiple mitigations we print:

```
Mitigations:
* Mitigation a
* Mitigation n
```

Reply:
Hm. I admit that this is more readable. At the expense of containing newline characters which are always a problem in logs.

Reply:
Changed.